javascript/lib.cpp

```diff
@@ -165,13 +165,16 @@ void CompiledIndex::Add(Napi::CallbackInfo const& ctx) {
 
     // Create an instance of the executor with the default number of threads
     auto run_parallel = [&](auto vectors) {
+        std::string error = "";
         executor_stl_t executor;
         executor.fixed(tasks, [&](std::size_t /*thread_idx*/, std::size_t task_idx) {
             add_result_t result = native_->add(static_cast<default_key_t>(keys[task_idx]),
                                                vectors + task_idx * native_->dimensions());
             if (!result)
-                Napi::Error::New(ctx.Env(), result.error.release()).ThrowAsJavaScriptException();
+                error += "<key:" + std::to_string(keys[task_idx]) + " message:" + result.error.release() + ">";
         });
+        if (error.size() > 0)
+            Napi::Error::New(ctx.Env(), error).ThrowAsJavaScriptException();
```

Comment:
This is not thread-safe. You need a `mutex` to safeguard the error string. Moreover, your solution may result in a heap allocation in the default string constructor. There should be a way to avoid it.

Reply:
Thanks for the review.
I understand that adding it to a string is not a good idea, but I would like to know if there is a better way to handle errors in such cases.

Would a Python implementation be helpful?
https://github.com/unum-cloud/usearch/blob/main/python/lib.cpp#L180-L211